src/movement_controls.py

```diff
@@ -0,0 +1,336 @@
+import arcade
+import math
+from player_input import PlayerInput
+
+
+import constants
+# This is setup for the Player to call "drive_input" where the vehicle updates it's
+# intended velocity and rotation based on the player input
+# then the player calls "movement" to act on the change from "drive_input"
+# this way the inputs are always registered but the movement can be skipped
+# if a collision or game state makes it where the vehicle shouldnt act
+# This will help avoid race conditions on collisions
+# 
+# 
+# for checking if a vehicle should take damage from an impact. as a percentage of their top acceleration
+
+DEFAULT_WORLD_SIZE = 750
```

Comment:
This may need to be increased.  When driving up against the right wall, the car teleports to the left edge of the screen.

Reply:
unrelated to the functionality, marking as resolved